Assets/Scripts/Interactables/InteractableObject.cs

```diff
@@ -0,0 +1,36 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[RequireComponent(typeof(Outline), typeof(SceneSwitcher))]
+public class InteractableObject : MonoBehaviour, IInteractable
+{
+    private Outline _outline;
+    private SceneSwitcher _sceneswitcher;
+    public string sceneName;
+
+    public void BeforeInteraction()
+    {
+        _outline.OutlineMode = Outline.Mode.OutlineVisible;
+    }
+
+    public void Interact()
+    {
+        _outline.OutlineMode = Outline.Mode.OutlineHidden;
+        StartCoroutine(_sceneswitcher.GoToNextScene(sceneName));
+    }
+    public void UndoBeforeInteraction()
+    {
+        _outline.OutlineMode = Outline.Mode.OutlineHidden;
+
+    }
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        _sceneswitcher = GetComponent<SceneSwitcher>();
+        _outline = GetComponent<Outline>();
+        _outline.OutlineMode = Outline.Mode.OutlineHidden;
```

Comment:
Set outline width to 0f since hidden mode don't actually hide outlines when object is behind another mesh (if i remember correctly?)